src/components/pc-block-field.vue

```diff
@@ -1,432 +1,275 @@
+    'suffix',
+  ],
+  data() {
+    return {
+      islockedFieldSet: (this.lockedField || '').length > 0,
+      isFocused: false,
+    }
+  },
+  updated() {
+    // Resync the editable fields
+    const el = this.$refs['pc-value']
+    if (el && el.textContent != this.formattedVal) {
+      el.textContent = this.formattedVal
+    }
 
-let validateFunctions = {
-        '*': {
-            // Base rate > 0
-            base: {
-                fn (value) {
-                    return value > 0
-                },
-                defaultVal: 10
-            },
-            // Power be between 0 and 100%(incl)
-            power: {
-                fn (value) {
-                    return value >= 0 && value <= 100
-                },
-                defaultVal: 80
-            },
-            days: {
-                fn (value) {
-                    return value > 0
-                },
-                defaultVal: 14
-            },
-            nonInfThreshold: {
-                fn (value) {
-                    return value > 0
-                },
-                defaultVal: 0
-            },
-            variants: {
-                fn (value) {
-                    return Number.isInteger(value) && value > 1
-                },
-                defaultVal: 1
-            }
-        },
-        gTest: {
-            // Limit Gtest rate input to be between 0.0001% and 99.999%
-            base: {
-                fn (value) {
-                    return value >= 0.01 && value <= 99.9
-                }
-            }
-        },
-        tTest: {
-            // Standard deviation > 0
-            sdRate: {
-                fn (value) {
-                    return value > 0
-                },
-                defaultVal: 10
-            }
-        }
+    // Once it updates, places the cursor back where it was. This is a visual
+    // glitch caused by the interaction between content editable and the prop drilling.
+    const focus = document.activeElement
+    if (el === focus) {
+      const range = document.createRange()
+      range.selectNodeContents(focus)
+      range.collapse(false)
+      const sel = window.getSelection()
+      sel.removeAllRanges()
+      sel.addRange(range)
+    }
+  },
+  computed: {
+    isLocked() {
+      return this.lockedField && this.lockedField == this.fieldProp
     },
-    validationCache = {};
+    formattedVal() {
+      let result = this.fieldValue
+      const sep = ','
 
-export default {
-    template: '#pc-block-field',
-    props: [
-        'lockedField',
-        'enableEdit',
-        'isReadOnly',
-        'fieldProp',
-        'fieldValue',
-        'prefix',
-        'suffix',
-        'fieldFromBlock'
-    ],
-    data () {
-        return {
-            islockedFieldSet: (this.lockedField || '').length > 0,
-            val: parseFloat(this.fieldValue),
-            isFocused: false,
+      if (result / 1000 >= 1) {
+        const [integer, decimal] = (result + '').split('.')
```

Comment:
Please use `String(result)` instead of adding it to an empty string

Reply:
Same as above.